src/components/PaginatedTable/index.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import classNames from 'classnames';
+import Button from '@/components/Button';
+
+import styles from '@/components/PaginatedTable/PaginatedTable.css';
+export interface Props {
+  currentPage: number;
+  handlePageChange: (value: number) => void;
+  headerList: React.ReactNode[];
+  numberOfItems: number;
+  numItemsPerPage: number;
+  tableContent: React.ReactNode[][];
+  variablesClassName?: string;
+}
+
+const PaginatedTable = (props: Props) => {
+  const {
+    currentPage,
+    handlePageChange,
+    headerList,
+    numberOfItems,
+    numItemsPerPage,
+    tableContent,
+    variablesClassName
+  } = props;
+
+  const numberOfPages = Math.ceil(numberOfItems / numItemsPerPage);
+
+  const handleChange = event => {
+    if (event.target.value >= 1 && event.target.value <= numberOfPages) {
+      handlePageChange(event.target.value);
+    }
+  };
+
+  const itemsRangeBegin = (currentPage - 1) * numItemsPerPage + 1;
+  const itemsRangeEnd = () => {
+    const total = itemsRangeBegin + numItemsPerPage;
+    return total > numberOfItems ? numberOfItems : total;
+  };
```

Comment:
Calculates the first and last pages for the items when you are navigating through the table pages.